cql3/expr/to_restriction.cc

```diff
@@ -20,15 +20,246 @@
  */
 
 #include "expression.hh"
+#include "cql3/statements/request_validations.hh"
+#include "cql3/restrictions/single_column_restriction.hh"
+#include "cql3/prepare_context.hh"
+#include "types/list.hh"
+#include <seastar/util/defer.hh>
+#include "exceptions/unrecognized_entity_exception.hh"
 
 namespace cql3 {
 namespace expr {
-::shared_ptr<restrictions::restriction> to_restriction(binary_operator& binop,
+namespace {
+logging::logger expr_to_restriction_logger("cql3_expr_to_restriction");
+
+bool is_legal_relation_for_non_frozen_collection(oper_t oper, bool is_lhs_col_indexed) {
+    return oper == oper_t::CONTAINS_KEY || oper == oper_t::CONTAINS || (oper == oper_t::EQ && is_lhs_col_indexed);
+}
```

Comment:
I took this validation code from here:
https://github.com/scylladb/scylla/blob/063d26bc9edd7d0ad1b1a32c849cf46eae60b098/cql3/single_column_relation.cc#L143

I will take a closer look at it.

Reply:
> I think you should move this one line of code into the one place that uses this function.

This is gonna make the if look ugly :/

In case of CONTAINS KEY `prepare_binary_operator` will check that the type is a map and fail if it is not.